File: dotnet-adapters-shortest-paths-example-project/main/roadrouting/CityRoadService.cs
using System.Collections.Generic;

namespace roadrouting {
    public interface CityRoadService {
	    IList<City> GetAllCities();
	    IList<Road> GetAllRoads();
	    City GetStartCity();
	    City GetEndCity();
	
	    /**
	     * Database implementation should close its resources
	     */
	    void ReleaseResourcesIfAny();
    }
}

File: dotnet-adapters-shortest-paths-example-project/main/java/roadrouting/CityRoadService.java
package roadrouting;

import java.util.List;

public interface CityRoadService {
	
	String  NAME_OF_START_CITY 	= "A";
	String  NAME_OF_END_CITY 	= "F";
	
	List<City> getAllCities();
	List<Road> getAllRoads();
	City getStartCity();
	City getEndCity();
	
	/**
	 * Database implementation should close its resources
	 */
	void releaseResourcesIfAny();
}
